android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -452,9 +362,154 @@ public void onSignalingChange(PeerConnection.SignalingState signalingState) {
+                    streamReactTag = e.getKey();
+                    break;
+                }
+            }
+            if (streamReactTag == null) {
+                streamReactTag = UUID.randomUUID().toString();
+                remoteStreams.put(streamReactTag, stream);
+            }
+            streams.pushMap(serializeStream(streamReactTag, stream));
+        }
+
+        eventInfo.putArray("streams", streams);
+        eventInfo.putMap("receiver", serializeReceiver(receiver));
+        eventInfo.putString("timestamp", ts);
+        // Getting the transceiver object associated with the receiver for the event
+        List<RtpTransceiver> transceivers = peerConnection.getTransceivers();
+        for( RtpTransceiver transceiver : transceivers ) {
+            if(transceiver.getReceiver() != null && receiver.id().equals(transceiver.getReceiver().id())) {
+                eventInfo.putMap("transceiver", serializeTransceiver(transceiver));
+                break;
+            }
+        }
+        params.putInt("id", this.id);
+        params.putMap("info", eventInfo);
+
+        webRTCModule.sendEvent("peerConnectionOnTrack", params);
+    }
+
+    /**
+     * Triggered when the signaling from SetRemoteDescription indicates that a transceiver
+     * will be receiving media from a remote endpoint. This is only called if UNIFIED_PLAN
+     * semantics are specified. The transceiver will be disposed automatically.
+     */
+    @Override
+    public void onTrack(final RtpTransceiver transceiver) {
+        Log.d(TAG, "onTrack");
+
+        RtpReceiver receiver = transceiver.getReceiver();
+        MediaStreamTrack track = receiver.track();
+
+        // Get The timestamp so that we can reorder the transceivers on the JS layer
+        Long tsLong = System.currentTimeMillis()/1000;
+        String ts = tsLong.toString();
+
+        if (track == null) {
+            return;
+        }
+        if(track.kind().equals(MediaStreamTrack.VIDEO_TRACK_KIND)){
+            videoTrackAdapters.addAdapter((VideoTrack) track);
+        }
+        remoteTracks.put(track.id(), track);
+
+        WritableMap params = Arguments.createMap();
+        WritableMap eventInfo = Arguments.createMap();
+        WritableArray streams = Arguments.createArray();
+        
+        eventInfo.putArray("streams", streams);
+        eventInfo.putMap("receiver", serializeReceiver(receiver));
+        eventInfo.putMap("transceiver", serializeTransceiver(transceiver));
+        eventInfo.putString("timestamp", ts);
+
+        // Getting the transceiver object associated with the receiver for the event
+        params.putInt("id", this.id);
+        params.putMap("info", eventInfo);
+
+        webRTCModule.sendEvent("peerConnectionOnTrack", params);
+    }
+
+    /*
+     * Triggered when a previously added remote track is removed by the remote
+     * peer, as a result of setRemoteDescription.
+     */
+    @Override
+    public void onRemoveTrack(RtpReceiver receiver){
```

Comment:
If the track is killed at this point we need to dispose of the adapter.

Reply:
Please verify what Chrome does.

Reply:
Verified in a comment above.